File: leetcode2/0100. Same Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if not p and not q:
            return True
        elif p and not q:
            return False
        elif not p and q:
            return False
        else:
            if p.val!=q.val:
                return False
            else:
                return self.isSameTree(q.left,p.left) and self.isSameTree(q.right,p.right)
        

File: archives/C#/0100. Same Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        if (p==null && q==null){
           return true;
        }
        else if((p==null && q!=null)||(p!=null && q==null)){
            return false;
        }
        else{
            if(p.val!=q.val){return false;}
            return IsSameTree(p.left,q.left) && IsSameTree(p.right,q.right);
        }
    }
}
